test/Base_Tests/src/Data/Decimal_Spec.enso

```diff
@@ -798,15 +798,15 @@ add_specs suite_builder =
             d.format format="#,##.##" locale=Locale.italy . should_equal "1.23.45.67.89,88"
 
         group_builder.specify "should parse correctly" <|
-            Decimal.parse "123456789.87654" . should_equal 123456789.87654
-            Decimal.parse "123,456,789.87654" locale=Locale.default . should_equal 123456789.87654
-            Decimal.parse "123,456,789.87654" locale=Locale.us . should_equal 123456789.87654
-            Decimal.parse "123.456.789,87654" locale=Locale.italy . should_equal 123456789.87654
+            Decimal.parse "123,456,789.87654" . should_equal (dec "123456789.87654")
+            Decimal.parse "123.456.789,87654" locale=Locale.italy . should_equal (dec "123456789.87654")
+            Decimal.parse "123,456,789.88" format="#,###.##" . should_equal (dec "123456789.88")
+            Decimal.parse "123.456.789,88" format="#,###.##" locale=Locale.italy . should_equal (dec "123456789.88")
```

Comment:
I'm not sure I understand how format and locale interact. This example looks to ignore the format and obey the locale?

Should we issue a warning in that case that we are ignoring format?

Reply:
I did some experimenting:

- It looks like the `format` parameter always uses "," as separator and "." as decimal point -- you can't express a format using localized characters, even when you specify the locale. You can't construct a `DecimalFormat` using `"###.###,##", regardless of locale.
- It looks like it doesn't really use many features of the format. If you specify "#,##.##" and give it a number with the usual three digits between commas, it parses it fine. It *does* respect the negative format, so you can specify `(123)` if you want. (I added some tests for this case.)

So I wouldn't say that it's ignoring the format because of the locale, it's more like it always ignores aspects of the pattern in all locales. This is barely documented in the javadocs, so I would imagine this parameter can only really be used by someone who knows that it's a Java format, and knows what they are doing.

Reply:
I was wrong -- you can use a localized pattern, but you must go through `NumberFormat`. The documentation has an example that suggests that sometimes you can't apply the pattern:

```
 NumberFormat f = NumberFormat.getInstance(loc);
 if (f instanceof DecimalFormat) {
     ((DecimalFormat) f).setDecimalSeparatorAlwaysShown(true);
 }
```

But I'll include an assertion that it can.